Docs no longer deploy properly

Looks like the action was based on calling `setup.py` this needs to be changed to using `sphinx-build` or equivalent.

Currently fails https://github.com/Becksteinlab/propkatraj/actions/runs/8919983113/job/24497145181

```
Run cd docs && sphinx-build -T -E --keep-going -b html -D language=en sphinx html
Running Sphinx v7.3.7

Traceback (most recent call last):
  File "/usr/share/miniconda3/envs/test/lib/python3.11/importlib/metadata/__init__.py", line 563, in from_name
    return next(cls.discover(name=name))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/share/miniconda3/envs/test/lib/python3.11/site-packages/sphinx/config.py", line 509, in eval_config_file
    exec(code, namespace)  # NoQA: S102
    ^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/propkatraj/propkatraj/docs/sphinx/conf.py", line 11, in <module>
    import propkatraj
  File "/home/runner/work/propkatraj/propkatraj/propkatraj/__init__.py", line 4, in <module>
    __version__ = version('propkatraj')
                  ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/miniconda3/envs/test/lib/python3.11/importlib/metadata/__init__.py", line 1009, in version
    return distribution(distribution_name).version
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/miniconda3/envs/test/lib/python3.11/importlib/metadata/__init__.py", line 982, in distribution
    return Distribution.from_name(distribution_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/miniconda3/envs/test/lib/python3.11/importlib/metadata/__init__.py", line 565, in from_name
    raise PackageNotFoundError(name)
importlib.metadata.PackageNotFoundError: No package metadata was found for propkatraj

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/share/miniconda3/envs/test/lib/python3.11/site-packages/sphinx/cmd/build.py", line 332, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/miniconda3/envs/test/lib/python3.11/site-packages/sphinx/application.py", line 211, in __init__
    self.config = Config.read(self.confdir, confoverrides or {}, self.tags)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/miniconda3/envs/test/lib/python3.11/site-packages/sphinx/config.py", line 309, in read
    namespace = eval_config_file(filename, tags)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/miniconda3/envs/test/lib/python3.11/site-packages/sphinx/config.py", line 522, in eval_config_file
    raise ConfigError(msg % traceback.format_exc()) from exc
sphinx.errors.ConfigError: There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/usr/share/miniconda3/envs/test/lib/python3.11/importlib/metadata/__init__.py", line 563, in from_name
    return next(cls.discover(name=name))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/share/miniconda3/envs/test/lib/python3.11/site-packages/sphinx/config.py", line 509, in eval_config_file
    exec(code, namespace)  # NoQA: S102
    ^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/propkatraj/propkatraj/docs/sphinx/conf.py", line 11, in <module>
    import propkatraj
  File "/home/runner/work/propkatraj/propkatraj/propkatraj/__init__.py", line 4, in <module>
    __version__ = version('propkatraj')
                  ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/miniconda3/envs/test/lib/python3.11/importlib/metadata/__init__.py", line 1009, in version
    return distribution(distribution_name).version
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/miniconda3/envs/test/lib/python3.11/importlib/metadata/__init__.py", line 982, in distribution
    return Distribution.from_name(distribution_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/miniconda3/envs/test/lib/python3.11/importlib/metadata/__init__.py", line 565, in from_name
    raise PackageNotFoundError(name)
importlib.metadata.PackageNotFoundError: No package metadata was found for propkatraj


Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/usr/share/miniconda3/envs/test/lib/python3.11/importlib/metadata/__init__.py", line 563, in from_name
    return next(cls.discover(name=name))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/share/miniconda3/envs/test/lib/python3.11/site-packages/sphinx/config.py", line 509, in eval_config_file
    exec(code, namespace)  # NoQA: S102
    ^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/propkatraj/propkatraj/docs/sphinx/conf.py", line 11, in <module>
    import propkatraj
  File "/home/runner/work/propkatraj/propkatraj/propkatraj/__init__.py", line 4, in <module>
    __version__ = version('propkatraj')
                  ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/miniconda3/envs/test/lib/python3.11/importlib/metadata/__init__.py", line 1009, in version
    return distribution(distribution_name).version
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/miniconda3/envs/test/lib/python3.11/importlib/metadata/__init__.py", line 982, in distribution
    return Distribution.from_name(distribution_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/miniconda3/envs/test/lib/python3.11/importlib/metadata/__init__.py", line 565, in from_name
    raise PackageNotFoundError(name)
importlib.metadata.PackageNotFoundError: No package metadata was found for propkatraj

Error: Process completed with exit code 2.
```

--- a/.github/workflows/docs.yaml
+++ b/.github/workflows/docs.yaml
@@ -18,24 +18,34 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
       with:
         fetch-depth: 0         
 
     - name: setup_miniconda
-      uses: conda-incubator/setup-miniconda@v2
+      uses: conda-incubator/setup-miniconda@v3
       with:
         python-version: 3.11
         environment-file: docs/requirements.yaml
-        auto-update-conda: true
-        channel-priority: flexible
-        channels: conda-forge
+        add-pip-as-python-dependency: true
+        architecture: x64
+
         miniforge-variant: Mambaforge
         use-mamba: true
+        channels: conda-forge
+
+        activate-environment: propkatrajdocs
+        auto-update-conda: true
+        auto-activate-base: false
+        show-channel-urls: true 
 
     - name: install dependencies
       run: |
         pip install pandas MDAnalysis propka==3.1
+
+    - name: install propkatraj
+      run: |
+        pip install .
 
     - name: build docs
       run: |

--- a/.github/workflows/docs.yaml
+++ b/.github/workflows/docs.yaml
@@ -3,6 +3,9 @@ on:
   push:
     branches:
       - main
+  pull_request:
+    branches:
+      - main
 
 concurrency:
   group: "${{ github.ref }}-${{ github.head_ref }}"
@@ -52,6 +55,7 @@ jobs:
         cd docs && sphinx-build -T -E --keep-going -b html -D language=en sphinx html
 
     - name: deploy docs
+      if: github.event_name != 'pull_request'    
       uses: peaceiris/actions-gh-pages@v3
       with:
         github_token: ${{ secrets.GITHUB_TOKEN }}

--- a/.github/workflows/gh-ci.yaml
+++ b/.github/workflows/gh-ci.yaml
@@ -3,11 +3,9 @@ on:
   push:
     branches:
       - main
-      - dev
   pull_request:
     branches:
       - main
-      - dev
   schedule:
     - cron: "0 0 * * 0"
 

--- a/docs/requirements.yaml
+++ b/docs/requirements.yaml
@@ -1,6 +1,6 @@
 name: propkatrajdocs
 channels:
-  - default
+  - conda-forge
 dependencies:
   - python
   - pip

